src/graphql_qb/queries.clj

```diff
@@ -95,28 +104,64 @@
         (str "FILTER(?ds = <" uri ">) ."))
       "}")))
 
-(defn get-unmapped-dimension-values-query [uri configuration]
-  (let [area-dim (config/geo-dimension configuration)
-        time-dim (config/time-dimension configuration)
-        codelist-label (config/codelist-label configuration)]
+(defn- process-dataset-bindings [bindings]
+  (-> bindings
+      (update :title util/label->string)
+      (update :description util/label->string)))
+
+(defn get-datasets [repo dimensions measures uri configuration lang]
+  (let [q (get-datasets-query dimensions measures uri configuration lang)
+        results (util/eager-query repo q)]
+    (map process-dataset-bindings results)))
+
+(defn- get-dataset-strings-query [dataset-uri configuration lang]
+  (let [label-predicate (str (config/dataset-label configuration))]
+    (str
+      "PREFIX qb: <http://purl.org/linked-data/cube#>"
+      "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>"
+      "SELECT ?title ?description WHERE {"
+      "  <" dataset-uri "> a qb:DataSet ."
+      "  OPTIONAL {"
+      "    <" dataset-uri "> <" label-predicate "> ?title ."
+      (when lang
+        (str "FILTER(LANG(?title) = \"" lang "\")"))
+      "  }"
+      "  OPTIONAL {"
+      "    <" dataset-uri "> rdfs:comment ?description ."
+      (when lang
+        (str "FILTER(LANG(?description) = \"" lang "\")"))
+      "  }"
+      "}")))
+
+(defn get-dataset-strings [repo dataset-uri configuration lang]
+  (let [q (get-dataset-strings-query dataset-uri configuration lang)
+        results (util/eager-query repo q)
+        bindings (first results)]
+    (process-dataset-bindings bindings)))
+
+(defn get-dimension-codelist-values-query [ds-uri configuration lang]
+  (let [codelist-label (config/codelist-label configuration)]
     (str
       "PREFIX qb: <http://purl.org/linked-data/cube#>"
       "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>"
       "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>"
       "PREFIX ui: <http://www.w3.org/ns/ui#>"
       "SELECT ?dim ?member ?label WHERE {"
-      "<" (str uri) "> qb:structure ?struct ."
+      "<" (str ds-uri) "> qb:structure ?struct ."
       "?struct a qb:DataStructureDefinition ."
       "?struct qb:component ?comp ."
-      "VALUES ?dim { <" (str area-dim) "> <" (str time-dim) "> }"
```

Comment:
You removed area-dim and time-dim from here but you use them at get-all-enum-dimension-values. I don't understand if CubiQL still handles geo and time dimension differently.

Reply:
They're treated specially in that we don't want to generate enum types for those dimensions because the codelists can be very large, which causes memory issues. We allow those dimensions to be queried differently by returning sub-properties of the dimension values e.g. URI, label etc.